pipeline/data_store.py

```diff
@@ -1306,65 +1310,100 @@ def save_and_commit( self, exists_ok=False, overwrite=True, no_archive=False,
             Note that this method calls session.commit()
 
         """
-        with SmartSession( session, self.session ) as session:
-            autoflush_state = session.autoflush
-            try:
-                # session.autoflush = False
-                for obj in self.get_all_data_products(output='list'):
-                    _logger.debug( f'save_and_commit consdering a {obj.__class__.__name__} with filepath '
-                                   f'{obj.filepath if isinstance(obj,FileOnDiskMixin) else "<none>"}' )
+        attributes = [
+            'exposure',
+            'image',
+            'sources',
+            'psf',
+            'wcs',
+            'zp',
+            'sub_image',
+            'detections',
+            'cutouts',
+            'measurements'
+        ]
```

Comment:
This is another one of those cases where if we change the code somewhere else to add a new thing to the datastore, we have to remember to add an element to this list.

At least, let's put a comment in __init__ that warns of this.  It might also be good to make a class atribute _attributes_to_save or some such that holds this list, which sits at the top of the whole class, and which the comment in __init__ points to.

Reply:
I've refactored it into two lists at the beginning of the class definition. I still chose to explicitly initialize them in `__init__` because then my IDE doesn't complain that these attributes are defined outside the initialization.